### PR TITLE
pass allocator to execute_distributed_partition

### PR DESCRIPTION
**Pull request by matthiasdiener**
Needs https://github.com/inducer/pytato/pull/280

### CODE REVIEW

File: grudge/array_context.py [hunk 1]
```diff
@@ -225,6 +225,11 @@ class MPIPytatoArrayContextBase(MPIBasedArrayContext):
     def __init__(
             self, mpi_communicator, queue, *, mpi_base_tag, allocator=None
             ) -> None:
+
+        if allocator is None:
+            from warnings import warn
+            warn("No memory allocator specified, please pass one.")
+
```
Comment (kaushikcfd) [inline grudge/array_context.py:232]: Thanks, I wonder if it might be better to put this in PyOpenCLArrayContext and PytatoPyOpenCLArrayContext.

Comment (matthiasdiener) [reply grudge/array_context.py]: Done in [875de83](https://github.com/inducer/grudge/pull/240/commits/875de839c206b3b5e4690c391ed0eadb28be0adb)

File: grudge/array_context.py [hunk 2]
```diff
@@ -88,6 +88,17 @@ class PyOpenCLArrayContext(_PyOpenCLArrayContextBase):
...
     any, for now.)
     """
+    def __init__(self, queue: "pyopencl.CommandQueue",
+            allocator: Optional["pyopencl.tools.AllocatorInterface"] = None,
+            wait_event_queue_length: Optional[int] = None,
+            force_device_scalars: bool = False) -> None:
+
+        if allocator is None:
+            from warnings import warn
+            warn("No memory allocator specified, please pass one.")
```
[the remark this answers is not in the document]
Comment (matthiasdiener) [reply grudge/array_context.py]: What do you think of e119659?

[the remark this answers is not in the document]
Comment (inducer) [reply grudge/array_context.py]: Sure, that works.

**Review by inducer — APPROVED** [review]

**Comment by matthiasdiener** [comment]
The flake8 error looks spurious:

```
examples/old_symbolics/dagrt-fusion.py:246:9: B020 Found for loop that reassigns the iterable it is iterating with each iterable value.
```

**Pull request merged**